clients/admin-ui/src/features/common/AccordionTree.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  Accordion,
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Box,
+  Button,
+  ButtonGroup,
+  Text,
+} from "@fidesui/react";
+import { Fragment, useState } from "react";
+
+import { TreeNode } from "./types";
+
+const ActionButtons = () => (
+  <ButtonGroup size="xs" isAttached variant="outline">
+    <Button>Edit</Button>
+    <Button>Delete</Button>
+  </ButtonGroup>
+);
+
+interface Props {
+  nodes: TreeNode[];
+}
+const AccordionTree = ({ nodes }: Props) => {
+  const [hoverNode, setHoverNode] = useState<TreeNode | undefined>(undefined);
+  /**
+   * Recursive function to generate the accordion tree
+   */
+  const createTree = (node: TreeNode) => {
+    // some nodes render as AccordionItems and some as just Boxes, but
+    // we want to keep their styling similar, so pass them the same props
+    const itemProps = {
+      // TODO: these borders are cut a little short, need to figure out CSS
```

Comment:
also would love some help with this one @ssangervasi if you have the chance... I'd like the border to go the full width even with padding. I thought `boxSizing="border-box"` would solve my problems, but alas, I must be doing something wrong! 

![image](https://user-images.githubusercontent.com/24641006/179301233-9b8eff04-08cd-4bc4-b05c-8c09559b8db3.png)

in this pic, the lines in between the entities should extend fully to the left 🤔

Reply:
Is there a way to apply a negative margin to the sub-items to extend beyond their width? From past experience, that's how I was able to accomplish something similar.

Reply:
I tried some negative margins today but couldn't quite get it to play nicely. The indents are coming from padding, so the negative margin just kind of undoes the padding. But I could very well be approaching this wrong—CSS is not my forte!

Reply:
I can only think of CSS approaches that require some very fancy selector-ing. Since we're already recursively building the elements with JS, my instinct is to pass a "level" parameter that increments on each recursive call. Then the margin for the children can be directly computed, while the border can be applied to a wrapper elements that have zero padding/margin.

Reply:
great idea, thanks! seemed to worked well https://github.com/ethyca/fides/pull/910/commits/c2a86634110e0bc500626d9622abf770626c8686